app/src/Login/SignupPage.tsx

```diff
@@ -8,23 +8,57 @@ export const SignupPage: React.FC = () => {
     const [psw1, setPsw1] = useState('');
     const [psw2, setPsw2] = useState('');
     const [isEmailInvalid, setIsEmailInvalid] = useState(false);
-    const [isPswInvalid, setIsPswInvalid] = useState(false);
-    const [pswHelperText, setPswHelperText] = useState(' ');
+    const [emailHelperText, setEmailHelperText] = useState('');
+    const [isPsw1Invalid, setIsPsw1Invalid] = useState(false);
+    const [psw1HelperText, setPsw1HelperText] = useState('');
+    const [isPsw2Invalid, setIsPsw2Invalid] = useState(false);
+    const [psw2HelperText, setPsw2HelperText] = useState('');
     const navigate = useNavigate();
 
     function handleSubmit() {
+        // Check email
         let parts = email.split('@');
         if (parts.length !== 2 || parts[1] !== 'uw.edu') {
             setIsEmailInvalid(true);
+            setEmailHelperText('Please use your @uw.edu email to sign up.');
+            return;
         } else {
             setIsEmailInvalid(false);
+            setEmailHelperText('');
         }
+        // Check psw1 has at least 8 characters
+        if (psw1.length < 8) {
+            setIsPsw1Invalid(true);
+            setPsw1HelperText('Password must be at least 8 characters long.');
+            return;
+        }
+        // Check psw1 has at least one capital letter
+        if (!psw1.match(/[A-Z]/)) {
```

Comment:
Can you make the inside of match() a constant at the top and then reference them for all of these?